rosbag2_cpp/src/rosbag2_cpp/message_definitions/local_message_definition_source.cpp

```diff
@@ -114,53 +121,56 @@ static std::string delimiter(const DefinitionIdentifier & definition_identifier)
   }
-  result += definition_identifier.package_resource_name;
+  result += definition_identifier.type_name();
   result += "\n";
   return result;
 }
 
-MessageSpec::MessageSpec(Format format, std::string text, const std::string & package_context)
-    : dependencies(parse_dependencies(format, text, package_context))
-    , text(std::move(text))
-    , format(format)
+LocalMessageDefinitionSource::MessageSpec::MessageSpec(
+  Format format, std::string text,
+  const std::string & package_context)
+: dependencies(parse_definition_dependencies(format, text, package_context))
+  , text(std::move(text))
+  , format(format)
 {
 }
 
-const MessageSpec & MessageDefinitionCache::load_message_spec(
+const LocalMessageDefinitionSource::MessageSpec & LocalMessageDefinitionSource::load_message_spec(
   const DefinitionIdentifier & definition_identifier)
 {
   if (auto it = msg_specs_by_definition_identifier_.find(definition_identifier);
-      it != msg_specs_by_definition_identifier_.end()) {
+    it != msg_specs_by_definition_identifier_.end())
+  {
     return it->second;
   }
   std::smatch match;
-  if (!std::regex_match(definition_identifier.package_resource_name, match,
-                        PACKAGE_TYPENAME_REGEX)) {
-    throw std::invalid_argument("Invalid package resource name: " +
-                                definition_identifier.package_resource_name);
+  const auto type_name = definition_identifier.type_name();
+  if (!std::regex_match(type_name, match, PACKAGE_TYPENAME_REGEX)) {
+    throw std::invalid_argument("Invalid type_name: " + type_name);
   }
   std::string package = match[1];
   std::string share_dir = ament_index_cpp::get_package_share_directory(package);
   std::ifstream file{share_dir + "/msg/" + match[2].str() +
-                     extension_for_format(definition_identifier.format)};
+    extension_for_format(definition_identifier.format())};
   if (!file.good()) {
-    throw DefinitionNotFoundError(definition_identifier.package_resource_name);
+    throw DefinitionNotFoundError(definition_identifier.type_name());
   }
 
   std::string contents{std::istreambuf_iterator(file), {}};
   const MessageSpec & spec =
     msg_specs_by_definition_identifier_
-      .emplace(definition_identifier,
-               MessageSpec(definition_identifier.format, std::move(contents), package))
-      .first->second;
+    .emplace(
+    definition_identifier,
+    MessageSpec(definition_identifier.format(), std::move(contents), package))
+    .first->second;
 
   // "References and pointers to data stored in the container are only invalidated by erasing that
   // element, even when the corresponding iterator is invalidated."
   return spec;
 }
 
-std::pair<Format, std::string> MessageDefinitionCache::get_full_text(
-  const std::string & root_package_resource_name)
+rosbag2_storage::MessageDefinition LocalMessageDefinitionSource::get_full_text(
+  const std::string & root_type_name)
```

Comment:
@james-rms Please rename parameter to avoid misleading in API
```suggestion
rosbag2_storage::MessageDefinition LocalMessageDefinitionSource::get_full_text(
  const std::string & root_topic_type)
```